server/client_fake.go

```diff
@@ -39,13 +39,18 @@ func (client *ClientFake) NodeSendStats(nodeKey string, sessionStats []dto.Sessi
 
 func (client *ClientFake) FindProposals(nodeKey string) (proposals []dto_discovery.ServiceProposal, err error) {
 	log.Info(MYSTERIUM_API_LOG_PREFIX, "Fake proposals requested for node_key: ", nodeKey)
-	if proposal, ok := client.proposalsByProvider[nodeKey]; ok {
-		proposals = []dto_discovery.ServiceProposal{proposal}
-	} else {
-		proposals = []dto_discovery.ServiceProposal{}
+
+	for _, proposal := range client.proposalsMock {
+		var filterMatched = true
+		if nodeKey != "" {
```

Comment:
This line means that if we create a node with empty nodeKey and search for proposals with nodeKey="A", we would get nodes with nodeKey="" as well. Do we really need this line?